pkg/types/ovirt/defaults/machinepools.go

```diff
@@ -0,0 +1,34 @@
+package defaults
+
+import (
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/types/ovirt"
+)
+
+func setMachinePool(p *types.MachinePool) {
+	if p.Platform.Ovirt == nil {
+		p.Platform.Ovirt = &ovirt.MachinePool{}
+	}
+}
+
+func setDefaultAffinityGroups(p *ovirt.Platform, mp *types.MachinePool, agName string) {
+	if mp.Platform.Ovirt.AffinityGroupsNames == nil {
+		for _, ag := range p.AffinityGroups {
+			if ag.Name == agName {
+				mp.Platform.Ovirt.AffinityGroupsNames = []string{agName}
+			}
+		}
+	}
+}
+
+// SetControlPlaneDefaults sets the defaults for the ControlPlane Machines.
+func SetControlPlaneDefaults(p *ovirt.Platform, mp *types.MachinePool) {
+	setMachinePool(mp)
+	setDefaultAffinityGroups(p, mp, DefaultControlPlaneAffinityGroupName)
```

Comment:
It is surprising behavior to me that the afinity group of a machine pool gets set if and only if there is an affinity group specified with an exact name. At a minimum, this needs to be well documented in the `affinityGroups` field description in the machine pool type.

Reply:
Well I'm not sure I understood you, but this is only for the defaults.
The user can specify anything he wants basically